robot/resources/lib/python_keywords/http_gate.py

```diff
@@ -240,11 +241,19 @@ def upload_via_http_gate_curl(
         assert match, f"Expected {output} to match {error_pattern}"
         return ""
 
-    oid_re = re.search(r'"object_id": "(.*)"', output)
-    if not oid_re:
-        raise AssertionError(f'Could not find "object_id" in {output}')
-    return oid_re.group(1)
+    json_match = re.search(r'{.*}', output, re.DOTALL | re.MULTILINE)
```

Comment:
Why this matching if you can just parse the JSON and it will fail anyway if it's not JSON?

Reply:
Curently output contains a string with a mix of curl command output and json:

` % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n                                 Dload  Upload   Total   Spent    Left  Speed\n\n  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\n100  575k    0     0    0  575k      0  2884k --:--:-- --:--:-- --:--:-- 2879k\n100 1537k    0     0    0 1537k      0  1258k --:--:--  0:00:01 --:--:-- 1258k{\n\t"object_id": "CUf93a2T6NocdLud77CtZcbsEfgQNMHBjfKjXQ3zKSe1",\n\t"container_id": "kLJKURtdNXpGc31qVhMmyD4zsjcY2nMTVXdbDBC3CHw"\n}\n\n100 1537k  100   129    0 1537k     75   895k  0:00:01  0:00:01 --:--:--  895k\n`

So we have to extract json to parse it first.

Reply:
Added --silent option to curl in the pipeline. That fixes the output. So RE is not needed, and parsing JSON is enough now.